src/utils/daos_hdlr.c

```diff
@@ -1566,11 +1598,1197 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	rc = dfs_lookup(file_dfs->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
+	if (parent == NULL) {
+		fprintf(stderr, "dfs_lookup %s failed, %d \n", dir_name, rc);
+		errno = rc;
+		D_GOTO(out, rc);
+	} else {
+		/* Stat the path */
+		rc = dfs_stat(file_dfs->dfs, parent, name, buf);
+		if (rc) {
+			fprintf(stderr, "dfs_stat %s failed (%d %s)\n",
+				name, rc, strerror(rc));
+			errno = rc;
+		}
+	}
+out:
+	if (parent != NULL) {
+		rc = dfs_release(parent);
+		if (rc != 0) {
+			fprintf(stderr, "dfs_release %s failed with error %d\n",
+				dir_name, rc);
+		}
+	}
+	if (name != NULL)
+		free(name);
+	if (dir_name != NULL)
+		free(dir_name);
+	return rc;
+}
+
+static int
+file_lstat(struct file_dfs *file_dfs, const char *path, struct stat *buf)
+{
+	int rc = 0;
+
+	if (file_dfs->type == POSIX) {
+		rc = lstat(path, buf);
+	} else if (file_dfs->type == DAOS) {
+		rc = stat_dfs(file_dfs, path, buf);
+	} else {
+		fprintf(stderr, "File type not known, file=%s, type=%d\n",
+			path, file_dfs->type);
+	}
+	return rc;
+}
+
+static ssize_t
+read_dfs(struct file_dfs *file_dfs,
+	 const char *file,
+	 void *buf,
+	 size_t size)
+{
+	d_iov_t		iov;
+	d_sg_list_t	sgl;
+	daos_size_t	got_size;
+
+	d_iov_set(&iov, buf, size);
+	sgl.sg_nr = 1;
+	sgl.sg_iovs = &iov;
+	sgl.sg_nr_out = 1;
+
+	/* execute read operation */
+	d_iov_set(&iov, buf, size);
+	int rc = dfs_read(file_dfs->dfs, file_dfs->obj, &sgl,
+			  file_dfs->offset, &got_size, NULL); 
+	if (rc) {
+		fprintf(stderr, "dfs_read %s failed (%d %s)\n",
+			file, rc, strerror(rc));
+		errno = rc;
+		D_GOTO(out, rc);
+	}
+
+	/* update file pointer with number of bytes read */
+	file_dfs->offset += (daos_off_t)got_size;
+out:
+	return (ssize_t)got_size;
+}
+
+static ssize_t
+file_read(struct file_dfs *file_dfs, char *file,
+		void *buf, size_t size)
+{
+	ssize_t got_size = 0;
+
+	if (file_dfs->type == POSIX) {
+		got_size = read(file_dfs->fd, buf, size);
+	} else if (file_dfs->type == DAOS) {
+		got_size = read_dfs(file_dfs, file, buf, size);
+	} else {
+		got_size = 1;
+		fprintf(stderr, "File type not known: %s type=%d\n",
+			file, file_dfs->type);
+	}
+	return got_size;
+}
+
+static int
+closedir_dfs(DIR *_dirp)
+{
+	struct	dfs_daos *dirp	= (struct dfs_daos *)_dirp;
+	int	rc		= dfs_release(dirp->dir);
+
+	if (rc) {
+		fprintf(stderr, "dfs_release failed (%d %s)\n",
+			rc, strerror(rc));
+		errno = rc;
+		rc = 1;
+	}
+	free(dirp);
+	return rc;
+}
+
+static int
+file_closedir(struct file_dfs *file_dfs, DIR *dirp)
+{
+	int rc = 0;
+
+	if (file_dfs->type == POSIX) {
+		rc = closedir(dirp);
+	} else if (file_dfs->type == DAOS) {
+		rc = closedir_dfs(dirp);
+	} else {
+		rc = 1;
+		fprintf(stderr, "File type not known, type=%d\n", file_dfs->type);
+	}
+	return rc;
+}
+
+static int
+close_dfs(struct file_dfs *file_dfs, const char *file)
+{
+	int rc = dfs_release(file_dfs->obj);
+
+	if (rc) {
+		fprintf(stderr, "dfs_close %s failed (%d %s)\n",
+			file, rc, strerror(rc));
+		errno = rc;
+	}
```

Comment:
@daltonbohning well I actually want to return the error code from these functions now, so I just updated to return that without setting 1/-1, so it should return whatever errno was set to. The POSIX functions set -1 on error, and then set errno, so not sure the best way to be consistent when the dfs_ functions which return errno code on failure. I've just been updating to return errno code always on failure for POSIX in the file_ wrappers.

Reply:
@dsikich Yeah, I'm also not sure which is the best way. It seems we have to choose between either being consistent with DFS, or being consistent with POSIX.
I imagine this will also come up when we formally centralize these wrapper functions into DFS somehow.

Reply:
@daltonbohning yeah I guess my thinking was I can try and make it consistent with DFS since these functions are going to be put with DFS, but it doesn't really matter to me as long as we agree on one way

Reply:
@dsikich Maybe we can let it be for now, as long as the error handling is in place to begin with. And then we can formally decide as part of [DAOS-6098](https://jira.hpdd.intel.com/browse/DAOS-6098)